src/components/slider/slider-base.ts

```diff
@@ -0,0 +1,515 @@
+      : thumbPositionX - mouseX;
+
+    if (this.step) {
+      const stepDistance = scale * this.step;
+      const stepDistanceCenter = stepDistance / 2;
+
+      // If the thumb scale range (slider update) is less than a half step,
+      // the position stays the same.
+      const scaleXPositive = Math.abs(change);
+      if (scaleXPositive < stepDistanceCenter) {
+        return 0;
+      }
+
+      return Math.round(change / stepDistance) * this.step;
+    } else {
+      return change / scale;
+    }
+  }
+
+  private updateSlider(mouseX: number) {
+    if (this.disabled) {
+      return;
+    }
+
+    const increment = this.calculateTrackUpdate(mouseX);
+    if (this.activeThumb && increment !== 0) {
+      this.updateValue(increment);
+    }
+  }
+
+  private pointerDown = (event: PointerEvent) => {
+    const thumb = this.closestHandle(event);
+    thumb.focus();
+
+    this.startValue = this.activeValue;
+    this.updateSlider(event.clientX);
+
+    this.setPointerCapture(event.pointerId);
+    this.pointerCaptured = true;
+    this.showThumbLabels();
+    event.preventDefault();
+  };
+
+  private pointerMove = (event: PointerEvent) => {
+    if (this.pointerCaptured) {
+      this.updateSlider(event.clientX);
+    }
+  };
+
+  private lostPointerCapture = () => {
+    this.pointerCaptured = false;
+    this.hideThumbLabels();
+
+    if (this.startValue! !== this.activeValue) {
+      this.emitChangeEvent();
+    }
+    this.startValue = undefined;
+  };
+
+  private handleKeydown = (event: KeyboardEvent) => {
+    if (this.disabled) {
+      return;
+    }
+
+    const { key } = event;
+
+    let increment = 0;
+    const value = this.activeValue;
+    const step = this.step ? this.step : 1;
+
+    switch (key) {
+      case 'ArrowLeft':
+        increment += this.isLTR ? -step : step;
+        break;
+      case 'ArrowRight':
+        increment += this.isLTR ? step : -step;
+        break;
+      case 'ArrowUp':
+        increment = step;
+        break;
+      case 'ArrowDown':
+        increment = -step;
+        break;
+      case 'Home':
+        increment = this.actualMin - value;
+        break;
+      case 'End':
+        increment = this.actualMax - value;
+        break;
+      case 'PageUp':
+        increment = Math.max((this.actualMax - this.actualMin) / 10, step);
+        break;
+      case 'PageDown':
+        increment = -Math.max((this.actualMax - this.actualMin) / 10, step);
+        break;
+      default:
+        return;
+    }
+
+    if (increment) {
+      const updated = this.updateValue(increment);
+      this.showThumbLabels();
+      this.hideThumbLabels();
+
+      if (updated) {
+        this.emitChangeEvent();
+      }
+    }
+  };
+
+  private handleThumbPointerEnter = () => {
+    this.showThumbLabels();
+  };
+
+  private handleThumbPointerLeave = () => {
+    this.hideThumbLabels();
+  };
+
+  protected renderTicks() {
+    const groups = [];
+    for (let i = 0; i < this.totalTickCount(); i++) {
```

Comment:
```suggestion
    for (let i = 0, totalCount = totalTickCount(); i < totalCount; i++) {
```
A little nitpick but I don't think we need to call the method on each iteration. Or maybe cache the tick count and update it when the ticks properties change.